artifacts/grammars/RCP-019/rcp019Lexer.g4

```diff
@@ -0,0 +1,121 @@
+lexer grammar rcp019Lexer;
+
+CONCAT: PIPE;
+LPAREN: '(';
+RPAREN: ')';
+SQUOTE: '\'';
+QUOTE: '"';
+DOT: '.';
+ASTERISK: '*';
+SLASH: '/';
+EXCLAMATION: '!';
+DOLLAR: '$';
+CARET: '^';
+BACKSLASH: '\\';
+SINGLE_SPACE: ' ';
+COLON: ':';
+PIPE: '|';
+LBRACKET: '[';
+RBRACKET: ']';
+HASH: '#';
+
+OR: '.OR.';
+AND: '.AND.';
+NOT: '.NOT.';
+
+EQ: '=';
+NE: EXCLAMATION EQ;
+LT: '<';
+LTE: LT EQ;
+GT: '>';
+GTE: GT EQ;
+
+CONTAINS: '.CONTAINS.';
+IN: '.IN.';
+COMMA: ',';
+PLUS: '+';
+MINUS: '-';
+MOD: '.MOD.';
+
+IIF: 'IIF';
+MATCH: 'MATCH';
+LAST: 'LAST';
+LIST: 'LIST';
+SET: 'SET';
+DIFFERENCE: 'DIFFERENCE';
+INTERSECTION: 'INTERSECTION';
+UNION: 'UNION';
+TRUE: 'TRUE';
+FALSE: 'FALSE';
+EMPTY: 'EMPTY';
+NULL: 'NULL';
+TODAY: 'TODAY';
+NOW: 'NOW';
+ENTRY: 'ENTRY';
+OLDVALUE: 'OLDVALUE';
+UPDATEACTION: 'UPDATEACTION';
+ANY: 'any';
+
+// See: Data Dictionary Member and Office Resources
+MEMBER_LOGIN_ID: 'MEMBER_LOGIN_ID';
+MEMBER_MLS_SECURITY_CLASS: 'MEMBER_MLS_SECURITY_CLASS';
+MEMBER_TYPE: 'MEMBER_TYPE';
+MEMBER_MLS_ID: 'MEMBER_MLS_ID';
+OFFICE_BROKER_MLS_ID: 'OFFICE_BROKER_MLS_ID';
+OFFICE_MLS_ID: 'OFFICE_MLS_ID';
+
+ALPHA: ('a' ..'z' | 'A' ..'Z');
+DIGIT: ('0' ..'9');
+
+// special tokens
+RESO_SPECIAL_TOKENS: FIELD_NAME | SPECOP;
+
+// TODO: Dynamically fill in your FIELD_NAMEs here
+FIELD_NAME:
+	'ListPrice'
+	| 'Status'
+	| 'CloseDate'
+	| 'Bedrooms'
+	| 'Bathrooms';
+
+SPECFUNC: IIF | MATCH;
+
+SPECOP:
+	EMPTY
+	| TRUE
+	| FALSE
+	| TODAY
+	| NOW
+	| ENTRY
+	| OLDVALUE
+	| MEMBER_LOGIN_ID
+	| MEMBER_MLS_SECURITY_CLASS
+	| MEMBER_TYPE
+	| MEMBER_MLS_ID
+	| OFFICE_BROKER_MLS_ID
+	| OFFICE_MLS_ID
+	| UPDATEACTION
+	| ANY;
+
+ALPHANUM: ALPHA (ALPHA | DIGIT)*;
+
+QUOTED_TERM: QUOTE (~[\\"])*? QUOTE | SQUOTE (~[\\'])*? SQUOTE;
```

Comment:
Many languages have ways to escape characters within a string. Was it intentional not to support this?

The way it's written now, representing the string

```
They said, "This won't work!"
```

would require multiple concatenations since it includes both a single and a double quote – it can't be expressed as either one string contained in single quotes or one string contained in double quotes.

```
"They said, " || '"' || "This won't work!" || '"'
```

or

```
'They said, "This won' || "'" || 't work!'
```

Allowing some sort of string escaping would make representing this string look more like (this works in JavaScript, among many others)

```js
"They said, \"This won't work!\""
```

Though I would be curious how adding escaping into the spec *now* would affect existing implementations.

Reply:
Agreed the grammar should support escape characters, but this is more of a v2 thing since it doesn't already.